lib/puppet/settings.rb

```diff
@@ -452,6 +452,13 @@ def include?(name)
   # Prints the contents of a config file with the available config settings, or it
   # prints a single value of a config setting.
   def print_config_options
+    if Puppet::Util::Log.sendlevel?(:info)
+      Puppet::Util::Log.newdestination(:console)
+      message = (_("Using --configprint is deprecated. Use 'puppet config <subcommand>' instead."))
+      Puppet.deprecation_warning(message)
+    end  
+    
+
```

Comment:
nit, there's some extra white space here. If you run `git show --check` you'll see it.